refreshVersions not supporting Gradle 9.0.0 because of `DefaultClientModule` dependency

- [ X ] I have read the [guidelines for submitting issues](https://splitties.github.io/refreshVersions/contributing/submitting-issues/#bug-reports)
## 🐛 Describe the bug

After upgrading to Gradle 9.0.0 running any gradle build with the refreshVersions plugin defined will result in an error and the build will fail immediately

## ⚠️ Current behavior

Running ./gradlew help results in:

```
FAILURE: Build failed with an exception.

* What went wrong:
org/gradle/api/internal/artifacts/dependencies/DefaultClientModule
> org.gradle.api.internal.artifacts.dependencies.DefaultClientModule

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s
```

## ✅ Expected behavior

I would expect the command to run normally. After removing refreshVersions completely, I get the following result, which is wat is to be expected.

Running ./gradlew help results in:

```

> Task :help

Welcome to Gradle 9.0.0.

To run a build, run gradlew <task> ...

To see a list of available tasks, run gradlew tasks

To see more detail about a task, run gradlew help --task <task>

To see a list of command-line options, run gradlew --help

For more detail on using Gradle, see https://docs.gradle.org/9.0.0/userguide/command_line_interface.html

For troubleshooting, visit https://help.gradle.org

BUILD SUCCESSFUL in 7s
1 actionable task: 1 executed
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.0.0/userguide/configuration_cache_enabling.html
```

## 💣 Steps to reproduce

* Upgrade to Gradle 9.0.0
* Add the refreshVersions plugin following the refreshVersions instructions
* Run `./gradlew help`

## 📱 Tech info
 - [Build Scan URL](https://dev.to/jmfayard/the-one-gradle-trick-that-supersedes-all-the-others-5bpg): <!-- e.g. Nexus One -->

--- a/plugins/convention-plugins/src/main/kotlin/PropertyOrEnv.kt
+++ b/plugins/convention-plugins/src/main/kotlin/PropertyOrEnv.kt
@@ -6,6 +6,10 @@ internal fun Project.propertyOrEnv(key: String): String {
         ?: error("Didn't find any value for the key \"$key\" in Project properties or environment variables.")
 }
 
-internal fun Project.propertyOrEnvOrNull(key: String): String? {
-    return findProperty(key) as String? ?: System.getenv(key)
+internal fun Project.propertyOrEnvOrNull(
+    key: String,
+    nullIfEmpty: Boolean = true
+): String? {
+    return (findProperty(key) as String?)?.takeUnless { nullIfEmpty && it.isEmpty() }
+        ?: System.getenv(key)?.takeUnless { nullIfEmpty && it.isEmpty() }
 }

--- a/plugins/convention-plugins/src/main/kotlin/gradle-plugin.gradle.kts
+++ b/plugins/convention-plugins/src/main/kotlin/gradle-plugin.gradle.kts
@@ -4,6 +4,7 @@ plugins {
 }
 
 signing {
+    isRequired = false
     useInMemoryPgpKeys(
         propertyOrEnvOrNull("GPG_key_id"),
         propertyOrEnvOrNull("GPG_private_key") ?: return@signing,

--- a/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCoreSetup.kt
+++ b/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCoreSetup.kt
@@ -15,7 +15,6 @@ import de.fayard.refreshVersions.core.internal.setupVersionPlaceholdersResolving
 import org.gradle.api.artifacts.ExternalDependency
 import org.gradle.api.file.RegularFile
 import org.gradle.api.initialization.Settings
-import org.gradle.api.internal.artifacts.dependencies.DefaultClientModule
 import org.gradle.kotlin.dsl.apply
 import org.gradle.tooling.UnsupportedVersionException
 import org.gradle.util.GradleVersion
@@ -212,7 +211,7 @@ private fun setupPluginsVersionsResolution(
                 val pluginVersion = requested.version ?: return@eachPlugin
                 UsedPluginsTracker.pluginHasNoEntryInVersionsFile(
                     settings = settings,
-                    dependency = pluginIdToDependency(pluginId, pluginVersion)
+                    dependency = pluginIdToDependency(settings, pluginId, pluginVersion)
                 )
                 return@eachPlugin
             }
@@ -248,5 +247,5 @@ internal fun pluginDependencyNotationToVersionKey(dependencyNotation: String): S
         else -> null
     }
 
-internal fun pluginIdToDependency(pluginId: String, version: String): ExternalDependency =
-    DefaultClientModule(pluginId, "$pluginId.gradle.plugin", version)
+internal fun pluginIdToDependency(settings: Settings, pluginId: String, version: String): ExternalDependency =
+    settings.gradle.rootProject.dependencies.create("$pluginId:$pluginId.gradle.plugin:$version") as ExternalDependency

--- a/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCoreSetup.kt
+++ b/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCoreSetup.kt
@@ -253,5 +253,5 @@ internal fun pluginIdToDependency(
     version: String
 ): ExternalDependency {
     val dependencyNotation = "$pluginId:$pluginId.gradle.plugin:$version"
-    return settings.gradle.rootProject.dependencies.create(dependencyNotation) as ExternalDependency
+    return settings.gradle.rootProject.buildscript.dependencies.create(dependencyNotation) as ExternalDependency
 }

--- a/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCoreSetup.kt
+++ b/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCoreSetup.kt
@@ -247,5 +247,11 @@ internal fun pluginDependencyNotationToVersionKey(dependencyNotation: String): S
         else -> null
     }
 
-internal fun pluginIdToDependency(settings: Settings, pluginId: String, version: String): ExternalDependency =
-    settings.gradle.rootProject.dependencies.create("$pluginId:$pluginId.gradle.plugin:$version") as ExternalDependency
+internal fun pluginIdToDependency(
+    settings: Settings,
+    pluginId: String,
+    version: String
+): ExternalDependency {
+    val dependencyNotation = "$pluginId:$pluginId.gradle.plugin:$version"
+    return settings.gradle.rootProject.dependencies.create(dependencyNotation) as ExternalDependency
+}

